synapse/storage/databases/main/roommember.py

```diff
@@ -908,42 +908,41 @@ async def _get_joined_users_from_context(
             ev_entry = event_map.get(event_id)
             if ev_entry and not ev_entry.event.rejected_reason:
                 if ev_entry.event.membership == Membership.JOIN:
-                    users_in_room[ev_entry.event.state_key] = ProfileInfo(
-                        display_name=ev_entry.event.content.get("displayname", None),
-                        avatar_url=ev_entry.event.content.get("avatar_url", None),
-                    )
+                    users_in_room.add(ev_entry.event.state_key)
             else:
                 missing_member_event_ids.append(event_id)
 
         if missing_member_event_ids:
-            event_to_memberships = await self._get_joined_profiles_from_event_ids(
+            event_to_memberships = await self._get_user_ids_from_membership_event_ids(
                 missing_member_event_ids
             )
-            users_in_room.update(row for row in event_to_memberships.values() if row)
```

Comment:
Ah, the removal of the `if row` is causing exception on my server, because there is an event_id being passed in that isn't in `room_memberships`. I'm currently trying to track it down.

Reply:
Oh, no, this is actually because we're filtering out non-joins, which @cachedList will return as None

Reply:
Fixed by https://github.com/matrix-org/synapse/pull/13600

Reply:
Oops, thank you for fixing that so quick 🙏!

Reply:
I was excited to test the change on my server, so deployed it and saw everything light up in red :D

I am a bit scared that none of the tests spotted this....